src/main/java/org/pac4j/j2e/filter/CallbackFilter.java

```diff
@@ -67,6 +68,7 @@ protected void internalFilter(final HttpServletRequest request, final HttpServle
         final Client client = clients.findClient(context);
         logger.debug("client: {}", client);
         CommonHelper.assertNotNull("client", client);
+        CommonHelper.assertTrue(client instanceof IndirectClient, "only indirect clients are allowed on the callback url");
```

Comment:
Only indirect clients are allowed on the callback urls: enforce that as we now have direct clients as well.